validators/cgroup_validator_linux.go

```diff
@@ -45,27 +45,72 @@ func (c *CgroupsValidator) Name() string {
 
 const (
 	cgroupsConfigPrefix = "CGROUPS_"
-	unifiedMountpoint   = "/sys/fs/cgroup"
+	mountsFilePath      = "/proc/mounts"
 )
 
+// getUnifiedMountpoint checks if the default mount point is available.
+// If not, it parses the mounts file to find a valid cgroup mount point.
+func getUnifiedMountpoint(path string) (string, error) {
+	f, err := os.Open(path)
+	if err != nil {
+		return "", err
+	}
+	defer f.Close()
+	scanner := bufio.NewScanner(f)
+	var cgroupV1MountPoint string
+	for scanner.Scan() {
+		line := scanner.Text()
+		if !strings.Contains(line, "cgroup") {
+			continue
+		}
+		// example fields: `cgroup2 /sys/fs/cgroup cgroup2 rw,seclabel,nosuid,nodev,noexec,relatime 0 0`
+		fields := strings.Fields(line)
+		if len(fields) >= 3 {
+			switch fields[2] {
+			case "cgroup2":
+				// return the first cgroup v2 mount point directly
+				return fields[1], nil
+			case "cgroup":
+				// set the first cgroup v1 mount point only,
+				// and continue the loop to find if there is a cgroup v2 mount point
+				if len(cgroupV1MountPoint) == 0 {
+					cgroupV1MountPoint = fields[1]
+				}
```

Comment:
is it ok to just do this?
```suggestion
				cgroupV1MountPoint = fields[1]
```

i.e. always update the var if more cgroups v1 mount points are found or should we only catch the first one?
(i don't know the format)

Reply:
```
[root@10-29-14-249 ~]# cat /proc/mounts | grep cgroup
tmpfs /sys/fs/cgroup tmpfs ro,seclabel,nosuid,nodev,noexec,mode=755 0 0
cgroup /sys/fs/cgroup/systemd cgroup rw,seclabel,nosuid,nodev,noexec,relatime,xattr,release_agent=/usr/lib/systemd/systemd-cgroups-agent,name=systemd 0 0
cgroup /sys/fs/cgroup/blkio cgroup rw,seclabel,nosuid,nodev,noexec,relatime,blkio 0 0
cgroup /sys/fs/cgroup/hugetlb cgroup rw,seclabel,nosuid,nodev,noexec,relatime,hugetlb 0 0
cgroup /sys/fs/cgroup/perf_event cgroup rw,seclabel,nosuid,nodev,noexec,relatime,perf_event 0 0
cgroup /sys/fs/cgroup/memory cgroup rw,seclabel,nosuid,nodev,noexec,relatime,memory 0 0
cgroup /sys/fs/cgroup/net_cls,net_prio cgroup rw,seclabel,nosuid,nodev,noexec,relatime,net_prio,net_cls 0 0
cgroup /sys/fs/cgroup/cpu,cpuacct cgroup rw,seclabel,nosuid,nodev,noexec,relatime,cpuacct,cpu 0 0
cgroup /sys/fs/cgroup/devices cgroup rw,seclabel,nosuid,nodev,noexec,relatime,devices 0 0
cgroup /sys/fs/cgroup/pids cgroup rw,seclabel,nosuid,nodev,noexec,relatime,pids 0 0
cgroup /sys/fs/cgroup/cpuset cgroup rw,seclabel,nosuid,nodev,noexec,relatime,cpuset 0 0
cgroup /sys/fs/cgroup/freezer cgroup rw,seclabel,nosuid,nodev,noexec,relatime,freezer 0 0
```

For cgroup v1, this is an example. I prefer to get the first one.